pkg/signup/service/signup_service.go

```diff
@@ -654,34 +643,27 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, username, phoneNumberOrHas
 //  2. the name of the default namespace
 //
 // If the user doesn't have access to any Space, then empty strings are returned
-func GetDefaultUserTarget(provider ResourceProvider, spaceName, murName string) (string, string) {
+func GetDefaultUserTarget(cl namespaced.Client, spaceName, murName string) (string, string) {
 	if spaceName == "" {
-		sbSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey, selection.Equals, []string{murName})
-		if err != nil {
-			log.Errorf(nil, err, "unable to create spacebindings selector for MUR %s", murName)
-			return "", ""
-		}
-
-		requirements := []labels.Requirement{*sbSelector}
-
-		sbs, err := provider.ListSpaceBindings(requirements...)
-		if err != nil {
+		sbs := &toolchainv1alpha1.SpaceBindingList{}
+		if err := cl.List(gocontext.TODO(), sbs, client.InNamespace(cl.Namespace),
```

Comment:
Not for this PR but it would be nice to propagate the original context here.

Reply:
Yes, I know. I decided to do it later as there are two other types of contexts used in the code - gin & echo - (plus the standard go context), and some of the code is called from both locations. Gin implements the standard go context interface, but echo doesn't. 
I have a story for this: [KUBESAW-199](https://issues.redhat.com/browse/KUBESAW-199)

Reply:
I can actually change the places where I can easily use the gin context, for the other places I would need to update the method signature which would introduce many other changes, so let me do it as a separate PR later
usage of the gin context (where it's easy to do) done here: 8187796